File: bdd/app/shopping_cart.rb
class ShoppingCart
  def initialize
    @content = []
  end
  
  def add(qty, product)
    @content << ShoppingCartItem.new(qty, product)
  end

  def total
    @content.sum(&:price)
  end
end

class ShoppingCartItem
  def initialize(qty, product)
    @qty = qty
    @product = product
  end

  def price
    @qty * @product.price
  end
end

File: behave/app/shoppingCart.py
class ShoppingCart(object):
  def __init__(self):
    self.items = []

  def add(self, qty, product):
    self.items.append(ShoppingCartItem(qty, product))

  def total(self):
    return sum(item.price() for item in self.items)

class ShoppingCartItem(object):
  def __init__(self, qty, product):
    self.qty = qty
    self.product = product

  def price(self):
    return self.qty * 10